presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleConnectorFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcIdentity;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+import oracle.jdbc.OracleConnection;
+import oracle.jdbc.OracleDriver;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class OracleConnectorFactory
+        implements ConnectionFactory
+{
+    private final String connectionUrl;
+    private final boolean connectionPoolEnabled;
+    private final int connectionPoolMinSize;
+    private final int connectionPoolMaxSize;
+    private final Properties connectionProperties;
+    private final CredentialProvider credentialProvider;
+
+    private PoolDataSource pds;
```

Comment:
Can we have an interface for `ConnectionProvider` and have implementation for for both pooledDataSource and a driver based one and bind them in the run time ?

Reply:
hi @Praveen2112, I change this factory to be only a pool factory..
for driver based I leave the DriverConnectorFactory that is already used..
this was a old code from previous PR.

if you could check this changes.